doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+        Downloading = 1,
+        /// The install is in progress. Cancellation of the IAsyncOperationWithProgress in this state will not stop the installation or the post install cleanup.
+        Installing = 2,
+        /// The installer has completed and cleanup actions are in progress. Cancellation of the IAsyncOperationWithProgress in this state will not stop cleanup or roll back the install.
+        PostInstall = 3,
+    };
+
+    /// Progress object for the install
+    /// DESIGN NOTE: percentage for the install as a whole is purposefully not included as there is no way to estimate progress when the installer is running.
+    [version(1)]
+    struct InstallProgress
+    {
+        /// State of the install
+        InstallProgressState State;
+        /// DESIGN NOTE: BytesDownloaded may only be available for downloads done by Windows Package Manager itself.
+        /// Number of bytes downloaded if known
+        UInt32 BytesDownloaded;
+        /// DESIGN NOTE: BytesRequired may only be available for downloads done by Windows Package Manager itself.
+        /// Number of bytes required if known
+        UInt32 BytesRequired;
+        /// Download percentage completed
+        UInt32 DownloadPercentage;
+    };
+
+    /// Result of the install
+    runtimeclass InstallResult
+    {
+        InstallResult();
+
+        /// Used by a caller to correlate the install with a caller's data.
+        String CorrelationId{ get; };
+        /// Whether a restart is required to complete the install.
+        Boolean RebootRequired { get; };
+    }
+
+    /// IMPLEMENTATION NOTE: SourceOrigin from AppInstallerRepositorySource.h
+    /// Defines the origin of the app catalog details.
+    enum AppCatalogOrigin
+    {
+        Default,
+        /// User means it was added by the user and could be removed.
+        User,
+        /// Predefined means it came as part of the Windows Package Manager package and cannot be removed.
+        Predefined,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceTrustLevel from AppInstallerRepositorySource.h
+    /// Defines the trust level of the app catalog.
+    enum AppCatalogTrustLevel
+    {
+        None,
+        Trusted,
+    };
+
+    /// IMPLEMENTATION NOTE: SourceDetails from AppInstallerRepositorySource.h
+    /// Interface for retrieving information about an app catalog without acting on it.
+    [version(1)]
+    struct AppCatalogDetails
+    {
+        /// The name of the app catalog.
+        String Name;
+        /// The type of the app catalog.
+        String Type;
+        /// The argument used when adding the app catalog.
+        String Arg;
+        /// The app catalog's extra data string.
+        String Data;
+        /// The app catalog's unique identifier.
+        String Identifier;
+        /// The last time that this app catalog was updated.
+        Windows.Foundation.DateTime LastUpdateTime;

```

Comment:
What is this set to when it's first added?  Do not use sentinel values for `DateTime`

Reply:
It gets set to std::chrono::system_clock::now() during the add.